libr/core/cmd_search.c

```diff
@@ -629,29 +629,50 @@ R_API RList *r_core_get_boundaries_prot(RCore *core, int protection, const char
 		if (m) {
 			append_bound (list, core->io, search_itv, m->itv.addr, m->itv.size);
 		}
-	} else if (!strcmp (mode, "io.maps")) { // Non-overlapping RIOMap parts not overriden by others (skyline)
+	} else if (!strncmp (mode, "io.maps", sizeof ("io.maps") - 1)) { // Non-overlapping RIOMap parts not overriden by others (skyline)
 		const RPVector *skyline = &core->io->map_skyline;
 		ut64 begin = UT64_MAX;
 		ut64 end = UT64_MAX;
 		size_t i;
+		int mask = 0;
+		bool readonly = false;
+
+		if (!strcmp (mode, "io.maps.exec")) {
```

Comment:
Can I express sections that are both executable and writable?

Reply:
i think this should go into a separate PR, but i agree that current approach have some limitations